crates/common/fork_choice/src/handlers.rs

```diff
@@ -145,10 +141,24 @@ pub async fn on_block(
     let block = &signed_block.message;
 
     // Parent block must be known
-    ensure!(store.block_states.contains_key(&block.parent_root));
+    if !store.block_states.contains_key(&block.parent_root) {
+        println!(
+            "Missing parent block state for parent_root: {:x}",
+            block.parent_root
+        );
+        return Ok(());
+    }
+
     // Blocks cannot be in the future. If they are, their consideration must be delayed until they
     // are in the past.
-    ensure!(store.get_current_slot() >= block.slot);
+    if store.get_current_slot() < block.slot {
+        println!(
+            "Block slot is ahead of current slot: block.slot = {}, store.get_current_slot() = {}",
+            block.slot,
+            store.get_current_slot()
+        );
+        return Ok(());
+    }
```

Comment:
```suggestion
ensure!(store.get_current_slot() >= block.slot, "Block slot is ahead of current slot: block.slot = {}, store.get_current_slot() = {}", block.slot, store.get_current_slot());
```

This is supposed to throw an error if it fails